src/pybind/mgr/cephadm/module.py

```diff
@@ -2510,8 +2507,19 @@ def generate_grafana_ds_config(hosts: List[str]) -> str:
         for dd in self.cache.get_daemons_by_service('prometheus'):
             prom_services.append(dd.hostname)
             deps.append(dd.name())
-        cert, pkey = create_self_signed_cert('Ceph', 'cephadm')
-        config_file = json.dumps({
+
+        # retrieve stored or generate (and store) TLS certificates
+        try:
+            cert = self.get_module_option_ex('orchestrator', 'grafana-tls-cert')
+            pkey = self.get_module_option_ex('orchestrator', 'grafana-tls-key')
```

Comment:
we want to use get_store, not get_module_option.. the store is in config-key instead of config (i.e., it won't appear with 'ceph config dump', but will with 'ceph config-key dump mgr/cephadm'.  there isn't an ``_ex`` version of get/set_store, so let's just use set_store/get_store for now and it will show up under mgr/cephadm/....